### PR TITLE
docs: Update the @deephaven/components docs

### PR DESCRIPTION
**Pull request by mofojed**
- Update the docs to actually have a working example

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -49,8 +49,8 @@
     "shortid": "^2.2.16"
   },
   "peerDependencies": {
-    "react": "^17.x",
-    "react-dom": "^17.x"
+    "react": "^16.8.0 || ^17.0.0-rc.1 || ^18.0.0",
+    "react-dom": "^16.8.0 || ^17.0.0-rc.1 || ^18.0.0"
```
Comment (mattrunyon) [inline packages/components/package.json:53]: Just use `>= 16.8.0`? No need for the 17 release candidates IMO

I use https://semver.npmjs.com/ to check what the actual published versions are that will match my version range

**Review by mattrunyon — CHANGES_REQUESTED** [review]
Should we update all our peer dependencies on react and react-dom?

File: packages/components/package.json (hunk 1 above)
Comment (mofojed) [reply packages/components/package.json:53]: Seems to be what other libs do: https://github.com/atlassian/react-beautiful-dnd/blob/013bfceac04ff48548c33cdc468dd2927446fc1b/package.json#L148
Though I think you're right, doesn't really seem necessary at this point...

Comment (mofojed) [reply packages/components/package.json]: I'm going to leave this in another change with #1493 and #1863 , I'll just update the docs with this PR.

**Review by mattrunyon — APPROVED** [review]

**Pull request merged**